src/parser/mod.rs

```diff
@@ -5416,7 +5416,8 @@ impl<'a> Parser<'a> {
             None
         };
 
-        let big_query_config = if dialect_of!(self is BigQueryDialect | GenericDialect) {
+        let big_query_config = if dialect_of!(self is BigQueryDialect | PostgreSqlDialect | GenericDialect)
```

Comment:
the `parse_optional_big_query_create_table_config()` function becomes a bit misleading with this change since its no longer bigquery specific (and would mean that we'd be supporting bigquery features for postgres which isn't necessariy desirable).
We can probably still reuse that function but make it dialect agnostic, e.g if we rename it to `parse_optional_create_table_config` and move the dialect_of checks into the function so that partitionby is guarded on both bigquery and postgres, while the rest are guarded on bigquery?

Reply:
@iffyio Good point. To be honest, I didn't notice this naming issue while adding the new dialect. I will rename and move the dialect_of guarded inside the function.

Reply:
@iffyio Done in the last commit, please take a look again when you get time ❤️